src/test/java/com/artipie/rpm/http/RpmSliceITCase.java

```diff
@@ -0,0 +1,148 @@
+import com.artipie.http.slice.LoggingSlice;
+import com.artipie.rpm.TestRpm;
+import com.artipie.vertx.VertxSliceServer;
+import io.vertx.reactivex.core.Vertx;
+import java.io.IOException;
+import java.util.Optional;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.text.StringContainsInOrder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Integration test for {@link RpmSlice}.
+ * @since 0.10
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+class RpmSliceITCase {
+
+    /**
+     * Vertx instance.
+     */
+    private static final Vertx VERTX = Vertx.vertx();
+
+    /**
+     * Vertx instance.
+     */
+    private VertxSliceServer slice;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Port.
+     */
+    private int port;
+
+    @Test
+    void installs() throws Exception {
+        this.start(Permissions.FREE, Identities.ANONYMOUS);
+        MatcherAssert.assertThat(
+            this.executeCommand(
+                String.format(
+                    "http://host.testcontainers.internal:%d/%s",
+                    this.port, "time-1.7-45.el7.x86_64.rpm"
+                )
+            ),
+            new StringContainsInOrder(new ListOf<>("time-1.7-45.el7.x86_64", "Complete!"))
+        );
+    }
+
+    @Test
+    void installsWithAuth() throws Exception {
+        final String john = "john";
+        final String pswd = "123";
+        this.start(
+            (name, perm) -> john.equals(name) && "download".equals(perm),
+            new BasicIdentities(
+                (name, pass) -> {
+                    final Optional<String> res;
+                    if (john.equals(name) && pswd.equals(pass)) {
+                        res = Optional.of(name);
+                    } else {
+                        res = Optional.empty();
+                    }
+                    return res;
+                }
+            )
+        );
+        MatcherAssert.assertThat(
+            this.executeCommand(
+                String.format(
+                    "http://%s:%s@host.testcontainers.internal:%d/%s", john,
+                    pswd, this.port, "time-1.7-45.el7.x86_64.rpm"
+                )
+            ),
+            new StringContainsInOrder(new ListOf<>("time-1.7-45.el7.x86_64", "Complete!"))
+        );
+    }
+
+    @AfterEach
+    void stopContainer() {
+        this.slice.close();
+        this.cntn.stop();
+    }
+
+    @AfterAll
+    static void close() {
+        RpmSliceITCase.VERTX.close();
+    }
+
+    private String executeCommand(final String url) throws IOException, InterruptedException {
+        return this.cntn.execInContainer(
+            "yum", "-y", "install", url
```

Comment:
@olenagerasimova I think this will download the file THEN install it. We should be using the url as a yum repository and not just a normal HTTP server in order to have yum read the metadata and everything.
I think the option `--repofrompath` is what should be used to specify the repository url.

Reply:
@victornoel 

> I think this will download the file THEN install it

Yes, and this is exactly what I want here: to verify that our `RpmSlice` can handle request coming from original rpm client, to make sure that our download api implementation is suitable for yum. 
But we indeed need the test you are suggesting here, don't you mind creating an issue?

Reply:
@olenagerasimova it is inconsequential that our Slice can answer random http request: if you need to do so, you don't need to use yum, you can just use any http client. Actually, you are just testing SliceDownload here. The project http already has tests for that.

The whole point of the todo (I wrote it) was to validate that this project can operate as a rpm repository with metadata and so on.

Reply:
@victornoel 

> it is inconsequential that our Slice can answer random http request: if you need to do so, you don't need to use yum, you can just use any http client

it's not a random http request, it's request yum sends to obtain rpm package (yes, this request is plain and simple). I'm testing here that two systems can interact with each other: yum can obtain package from our storage using our download api implementation. Yes, it is primitive - get requests.

Test you are proposing should be another integration test for `Rpm`: we need to create rpm repo with our `Rpm#batchUpdate` first, then configure yum repository with YUM repository configuration file and list/install packages from our this repository. This another rpm-adapter use-case.

Reply:
@olenagerasimova I don't know how to convince you. The todo is quite clear that the requirement here is to test the use of artipie as a repository and even give a SO link to explain how to setup the repository with yum. So if you insist on continuing with this current PR, please add a todo to finally test the desired behaviour of artipie as a yum repository.

You are ARC so you must know what you are doing, but if you weren't I would ask ARC to refuse the PR because it does not solve the todo (it's not far from doing it though, so adding a todo to go into that direction could also work).

Maybe we can ask a third party such as @g4s8 to interfere here?

Reply:
@victornoel let' check the todo:

> Add integration tests for RpmSlice to validate that it **can answer requests coming from the rpm original client**

in my test that's exactly what's happening: yum sends request to our slice to obtain the package. Also, in the sentence you are referring to `RpmSlice` directly, not to the rpm repository built by rpm-adapter. Slices in this repo have very simple purpose: download or upload packages. I have no idea what you meant, I just explained to you how and why I understand this todo.

I do not support the idea of closing this PR as I do not think this test is useless, but I'm proposing you to open another issue to suggest the test you are proposing here. From my prospective it's a win-win: you got reworded for the issue, this repository get more tests.

Reply:
@olenagerasimova please add a todo for creating an integration test where yum uses rpm-adapter as a repository. Don't forget to include the SO link that was in the previous todo. Thanks